Derek 94

Derek-94 opened the pull request:
First pull request _ made it without referring to other people's code.

Derek-94 committed:
feat(racingcar): manage car names

- input car names.
- judge if it is acceptable (should less than 5 letters).
- add ids on HTML files.

Derek-94 committed:
feat(main): all functions update

- added tryInput click event.
- added car class, also array.
- making random number function update.
- extract maximum race record function update
- added printing final result function.

Derek-94 committed:
Create README.md

-things we should build in list.

Derek-94 committed:
refactor: modularize printing function

- modularize printing function.
- make comment concise.

Derek-94 committed:
fix: delete unnecessary files

- delete index.js file that is not used

lxxjn0 commented on docs/README.md:
```diff
@@ -0,0 +1,31 @@
+# 🏎 자동차 경주 게임
```
Markdown 파일에서 

```
- [ ] 체크 안한 리스트

- [x] 체크한 리스트
```

- [x] 체크 안한 리스트

- [ ] 체크한 리스트

이런 식으로 작성하면 체크가 됩니다 👍 

그래서 주로 목록을 작성하고 체크하면서 커밋을 작성하기도 해요!! 이건 그냥 참고하시라고 코멘트로 남겨둘게요 :)

lxxjn0 commented on src/carMake.js:
```diff
@@ -0,0 +1,20 @@
+
+export default function makeCarObject(){
+    nameTokens.forEach(function(obj){
+        let tmp = new Car(obj);
+        carArray.push(tmp);
```
이 부분이 이해가 잘 안가요

lxxjn0 reviewed (changes requested):
🤩

JeongP commented on index.html:
```diff
@@ -14,19 +14,23 @@ <h1>🏎️ 자동차 경주 게임</h1>
+    <script type="module" src="src/carMake.js"></script>
+    <script type="module" src="src/tryCount.js"></script>
+    <script type="module" src="src/carRace.js"></script>
+    <script type="module" src="src/printResult.js"></script>
```
코드가 어디서 시작되는지를 잘 모르겠어서, 파일을 하나하나 다 열어봐야 하는 번거로움이 있는 것 같아요!
저의 경우는 index.js에서 생성자를 통해 어떻게 시작이 되는지 명시를 해줬는데, 참고해보시면 좋을 것 같습니다.🙃

JeongP reviewed (commented):
Project 1에 대한 전체적인 리뷰입니닷..(한개..)😅

Derek-94 committed:
refactor: revise according to convention

- make constant number and apply them
- change variables name clear
- make indention in function statement standard
- use arrow function in statements that can be shorten.

Derek-94 committed:
refactor: chagne names and type

- change variable name to make clear
- change fucntion name to cnt

Derek-94 committed:
fix: Fix some bugs

- Update try count page to comes up when user input car name.
- Fix the bug when user input too long car name. It cannot catch the exception properly.
- Add some CSS effects to fade in.

Derek-94 committed:
refactor: Modularize from one .js file.

- Update one big class that manages all the event Listener. (carRacingGameControl class.)
- Make index.js to modularize from one file.
- Make constants file that accumulate all the constant value in program.

Derek-94 committed:
fix: Remove unnecessarystatement.

- Remove export default statement in declaring functions.

Derek-94 committed:
refactor: Remove and rename some variables.

- Remove not used variable.
- Remove comments.
- Change function name to make its mean clear.
- Change type of variable to make it safe.
